support/maps/maps.go

```diff
@@ -25,6 +29,54 @@ func Forget[K comparable, V any](mp map[K]V, keys ...K) {
 	}
 }
 
+func FromStruct(data any) map[string]any {
+	res := make(map[string]any)
+	dataType := reflect.TypeOf(data)
+	dataValue := reflect.ValueOf(data)
+
+	if dataType.Kind() == reflect.Pointer {
+		dataType = dataType.Elem()
+		dataValue = dataValue.Elem()
+	}
+
+	if dataType.Kind() != reflect.Struct {
+		return res
+	}
+
+	for i := 0; i < dataType.NumField(); i++ {
+		fieldType := dataType.Field(i)
+		fieldValue := dataValue.Field(i)
+
+		if !fieldType.IsExported() {
+			continue
+		}
+
+		if fieldValue.Kind() == reflect.Pointer {
+			if fieldValue.IsNil() {
+				res[fieldType.Name] = nil
+				continue
+			}
+
+			fieldValue = fieldValue.Elem()
+		}
```

Comment:
Can be optimize to `fieldValue = reflect.Indirect(fieldValue)`

Reply:
If so, how to add:

```
if fieldValue.IsNil() {
  res[fieldType.Name] = nil
  continue
}
```